### PR TITLE
Update __shfl_down to __shfl_down_sync for compilation with CUDA 10.1

### PR DESCRIPTION
**Pull request by minitu**
This fixes undefined errors for `__shfl_down` when compiling with CUDA 10.1.
Tested on OLCF Summit.

### CODE REVIEW

File: HostCUDA.cu [hunk 1]
```diff
@@ -1572,11 +1572,11 @@ __global__ void nodeGravityComputation(
       poten = pot;
       idt2max = idt2;
       for (int offset = NODES_PER_BLOCK/2; offset > 0; offset /= 2) {
-        sumx += __shfl_down(sumx, offset, NODES_PER_BLOCK);
-        sumy += __shfl_down(sumy, offset, NODES_PER_BLOCK);
-        sumz += __shfl_down(sumz, offset, NODES_PER_BLOCK);      
-        poten += __shfl_down(poten, offset, NODES_PER_BLOCK);
-        idt2max = fmax(idt2max, __shfl_down(idt2max, offset, NODES_PER_BLOCK));
+        sumx += __shfl_down_sync(sumx, offset, NODES_PER_BLOCK);
```
Comment (trquinn) [inline HostCUDA.cu]: Looking at the documentation for __shfl_down_sync(), the first argument needs to be a thread mask, (e.g., 0xffffffff).  I'm guessing this is causing CUDA errors.

**Comment by hainest** [comment]
Doing a blanket replacement requires that we use at least CUDA 9.0 which we should enforce in the build system. Otherwise, we can wrap the `__shfl` instructions to support all versions.

**Comment by trquinn** [comment]
There is at least one XSEDE system (comet) which has CUDA 7 as the default environment, so it would be nice to have the wrappers.

**Comment by hainest** [comment]
@minitu I couldn't push to your branch, but I have a wrapper implementation [branch](https://github.com/hainest/changa/commit/e2e728337f57f3fbee27f02e7a5f9b9e5c07badd) that has been tested to build successfully with CUDA 7.5, 9.2, and 10.0. You should be able to cherry-pick that onto your branch. That fix also starts to lay the foundation for moving CUDA-specific code into it's own subdirectory in anticipation of eventually having ChaNGa support multiple GPU platforms.

**Comment by trquinn** [comment]
There is also the __any() intrinsic used in the GPU_LOCAL_TREE_WALK code.  Should we fix that here, or do it in another PR?

**Comment by hainest** [comment]
@trquinn I'm working on another PR for that. Oddly, I wasn't seeing a deprecation warning for that when building on CUDA 10. I know that all of the unsync ballot functions are deprecated, though.

**Pull request merged**